### PR TITLE
Preprocessor adblock rules and apply if-statements

### LINKED ISSUE #923: Remove unsupported preprocessor-based filters from an adblock list
We don't currently support [preprocessor directives](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#if-condition) in adblock-rust. This leads to issues when upstream lists push filters that are only meant for Firefox. We should filter out rules that are FF-only on the backend.

chrmod shared https://gist.github.com/chrmod/f26267a9fcd3f39d9adfd401f5c5b168 on our shared Slack which we can adapt pretty easily to filter out rules. We'd want to remove both `cap_html_filtering` and `env_firefox`.

### PR DESCRIPTION
**Pull request by bcaller**
This allows us to exclude firefox rules.
Closes #923

### CODE REVIEW

**Comment by bcaller** [comment]
Not sure how changing two javascript files has given me a rust compilation error in CI.

**Comment by ShivanKaul** [comment]
@antonok-edm looks like there's an adblock-rs related issue during npm install? https://github.com/brave/brave-core-crx-packager/actions/runs/9590654307/job/26446339704?pr=932

**Comment by antonok-edm** [comment]
@bcaller that issue should be fixed as of https://github.com/brave/brave-core-crx-packager/pull/934, can you rebase?

**Comment by bcaller** [comment]
Fixed, thanks

File: test/testAdBlock.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import path from 'path'
 import { spawnSync } from 'child_process'
 
 import { generateResourcesFile, sanityCheckList } from '../lib/adBlockRustUtils.js'
+import { preprocess } from '../scripts/generateAdBlockRustDataFiles.js'
```
Comment (antonok-edm) [inline test/testAdBlock.js:9]: could we move `preprocess` to `adBlockRustUtils`?

**Review by antonok-edm — APPROVED** [review]

File: test/testAdBlock.js (hunk 1 above)
Comment (bcaller) [reply test/testAdBlock.js]: done

**Comment by ShivanKaul** [comment]
@bcaller is this good to merge in now?

**Comment by bcaller** [comment]
yes

**Pull request merged**